jlm/llvm/opt/PredicateCorrelation.hpp

```diff
@@ -30,6 +31,93 @@ class StatisticsCollector;
 namespace jlm::llvm
 {
 
+/**
+ * The different types of predicate correlations that are supported.
+ */
+enum class CorrelationType
+{
+  /**
+   * The predicate correlates with control constants from subregions of another structural node.
+   */
+  ControlConstantCorrelation,
+};
+
+/**
+ * Represents the predicate correlation between a theta node and a gamma node that resides in the
+ * theta node's subregion.
+ */
+class ThetaGammaPredicateCorrelation final
+{
+public:
+  using ControlConstantAlternatives = std::vector<size_t>;
+  using CorrelationData = std::variant<ControlConstantAlternatives>;
+
+private:
+  ThetaGammaPredicateCorrelation(
+      const CorrelationType type,
+      rvsdg::ThetaNode & thetaNode,
+      rvsdg::GammaNode & gammaNode,
+      ControlConstantAlternatives controlConstantAlternatives)
```

Comment:
should this type not be `CorrelationData`?

Reply:
Fixed.